Allows install paths with spaces on MacOS

TimArt opened the pull request:
I've modified the MacOS build script to allow compilation in a directory hierarchy that includes directory names with spaces in them. I did this by using "weak quoting" in bash. I also added git submodule initialization and updating so when a user goes to build for MacOS they do not first need to know about git submoduling.

I've added a comment on each line I've changed to help explain. Some of these comments are unnecessary and will probably be removed.

Thanks!

thp commented:
Yeah, no need to add comments to each line. Remove the comments if possible. Also, maybe this is a good time to run it through the `shellcheck` utility (https://www.shellcheck.net, `brew install shellcheck` on macOS) and fix those things.

TimArt commented:
I removed my unnecessary comments and corrected the following lines based on `shellcheck`:
 
<img width="966" alt="screen shot 2018-02-19 at 11 56 08 am" src="https://user-images.githubusercontent.com/8052822/36395962-6a1e56a0-1581-11e8-95c6-c96390934bc1.png">

Let me know if there's anything I've done that you want modified.

thp commented:
Cool, thank you!
